core/src/main/java/io/grpc/internal/MessageDeframer.java

```diff
@@ -346,19 +349,15 @@ private InputStream getUncompressedBody() {
   }
 
   private InputStream getCompressedBody() {
-    if (compression == Compression.NONE) {
+    if (decompressor == MessageEncoding.NONE) {
```

Comment:
I guess I'm having trouble with the definition of NONE:
> Special sentinel codec indicating that no compression should be used.  Users should use reference equality to see if compression is disabled.

I guess it's fine. I think I see the reasons, but something just seems... off.